src/query/service/src/interpreters/interpreter_copy.rs

```diff
@@ -253,45 +250,42 @@ impl CopyInterpreter {
             Some(max_files)
         };
 
-        let mut all_source_file_infos = if force {
+        let all_source_file_infos = if force {
             StageTable::list_files(&stage_table_info, max_files).await?
         } else {
             StageTable::list_files(&stage_table_info, None).await?
         };
+        let num_all_files = all_source_file_infos.len();
 
-        info!("end to list files: {}", all_source_file_infos.len());
+        info!("end to list files: got {} files", num_all_files);
 
-        if !force {
+        let need_copy_file_infos = if force {
```

Comment:
Same as above, if user use max_files option, we assume he absolutely know what he did, it's not related with the force option.

Reply:
In the case he only wants force copy max_files files for test or others purposes.
Or, he shouldn't use the max_files.